crates/ruff_dev/src/format_dev.rs

```diff
@@ -209,99 +208,60 @@ pub(crate) fn main(args: &Args) -> anyhow::Result<ExitCode> {
 
-                let sender = sender.clone();
-
-                scope.spawn(move |_| {
-                    sender.send(Message::Start { path: path.clone() }).unwrap();
-
-                    match format_dev_project(
-                        &[path.clone()],
-                        args.stability_check,
-                        args.write,
-                        false,
-                    ) {
-                        Ok(result) => sender.send(Message::Finished { result, path }),
-                        Err(error) => sender.send(Message::Failed { error, path }),
-                    }
-                    .unwrap();
-                });
-            }
-        }
+    let num_projects = project_paths.len();
+    let bar = ProgressBar::new(num_projects as u64);
 
-        // Main thread, writing to stdout
-        #[allow(clippy::print_stdout)]
-        scope.spawn(|_| {
-            let mut error_file = args.error_file.as_ref().map(|error_file| {
-                BufWriter::new(File::create(error_file).expect("Couldn't open error file"))
-            });
+    let mut error_file = args.error_file.as_ref().map(|error_file| {
+        BufWriter::new(File::create(error_file).expect("Couldn't open error file"))
+    });
 
-            let bar = ProgressBar::new(num_projects as u64);
-            for message in receiver {
-                match message {
-                    Message::Start { path } => {
-                        bar.suspend(|| println!("Starting {}", path.display()));
-                    }
-                    Message::Finished { path, result } => {
-                        total_errors += result.error_count();
-                        total_files += result.file_count;
-
-                        bar.suspend(|| {
-                            println!(
-                                "Finished {} with {} files (similarity index {:.3}) in {:.2}s",
-                                path.display(),
-                                result.file_count,
-                                result.statistics.similarity_index(),
-                                result.duration.as_secs_f32(),
-                            );
-                        });
-                        bar.suspend(|| print!("{}", result.display(args.format)));
-                        if let Some(error_file) = &mut error_file {
-                            write!(error_file, "{}", result.display(args.format)).unwrap();
-                            error_file.flush().unwrap();
-                        }
-                        bar.println(result.display(args.format).to_string().trim_end());
-                        bar.inc(1);
-                    }
-                    Message::Failed { path, error } => {
-                        bar.suspend(|| println!("Failed {}: {}", path.display(), error));
-                        bar.inc(1);
-                    }
+    for project_path in project_paths {
+        bar.println(format!("Starting {}", project_path.display()));
```

Comment:
```suggestion
        bar.suspend(|| println!("Starting {}", project_path.display()));
```

otherwise it won't print when redirecting to a file (https://docs.rs/indicatif/latest/indicatif/struct.ProgressBar.html#method.println, such a footgun), same below

Reply:
oh man. And I thought it is because it would mess up the bar... I'll change it back